### DISCUSSION TITLE
How to handle multiple databases in the same project

### DISCUSSION THREAD

**Discussion by haikyuu** [geldata/gel#2902] [Q&A] [answered]
I want to add e2e tests to my project, and I want to know how to use a different database for tests?

Should I create a new edgedb project and give it a different root folder path? Or is there a better way?

I would actually prefer creating a test_db database, and be able to point the current edgedb project to it using environment variables. That would be super neat as I won't have to change anything in my code base.

One more thing, is there some documentation for the `edgedb.toml` file?

**Comment by elprans** [comment, marked as answer]
Use the `EDGEDB_DATABASE` environment variable to specify which database inside an instance to connect to.

> One more thing, is there some documentation for the `edgedb.toml` file?

`edgedb.toml` is currently only used as an `edgedb project` marker and can specify the version of EdgeDB server to be used with the project:

```
[edgedb]
server-version = "1-beta2" # can also be "nightly"
```

We expect that you wouldn't normally need to modify the file directly, it's usually created and modified by `edgedb project init` and `edgedb project upgrade`.

**Reply by haikyuu** [reply]
Thanks @elprans

**Reply by haikyuu** [reply]
@elprans It's not picked up by `createPool()` nor `edgedb()`.

And I tried setting the dsn manually, but I don't seem to get it right ... as the following keeps hanging

```sh
$ edgedb --dsn "edgedb://edgedb@localhost/test"  
Connecting to an EdgeDB instance at localhost:5656...
```

**Reply by CodesInChaos** [reply]
1. I'd use a named instance and `-I` for the alternative db (but that doesn't really affect your problem)
2. Is the server running? `instance list` should show the status of all instances, and `instance start <name>` should start it if it's not running

**Reply by haikyuu** [reply]
@CodesInChaos yes the server is running. And I get the port when issuing `instance list`. But then I need to specify the password in the dsn as well ...

I ended up manually adding the database name in the code. Specifying the instance name isn't required.
```
db = await edgedb.createPool(process.env.EDGEDB_DATABASE === "test" ? {database:"test"}: undefined)
```

It would be nice if the edgedb-js client would do this automagically.

One gotcha is that createPool connectionOptions complains about database not being a valid option. I'll open an issue for it.

**Comment by Extarys** [comment]
Is this still the recommended way in 2025 with geldata ? I think a lot changed under the hood. :thinking:
